internal/status_test.go

```diff
@@ -1,12 +1,47 @@
 package internal
 
 import (
+	"context"
+	"fmt"
 	"testing"
+	"time"
 
+	"github.com/cisco-open/grabit/test"
 	"github.com/stretchr/testify/assert"
 )
 
-func TestComposeStatusString(t *testing.T) {
-	// TODO
-	assert.True(t, true)
+func TestGetStatusString(t *testing.T) {
+	// Test 1000 resources.
+	content := `abcdef`
+	port := test.TestHttpHandler(content, t)
+	resources := []Resource{}
+	algo := "sha256"
+	for i := 0; i < 1000; i++ {
+		resource := Resource{Urls: []string{fmt.Sprintf("http://localhost:%d/test%d.html", port, i)}, Integrity: fmt.Sprintf("%s-vvV+x/U6bUC+tkCngKY5yDvCmsipgW8fxsXG3Nk8RyE=", algo), Tags: []string{}, Filename: ""}
+		resources = append(resources, resource)
+	}
+	ctx, _ := context.WithCancel(context.Background())
+
+	// Test resource generation.
+	assert.Len(t, resources, 1000)
+
+	// Test StatusLine initialization and initResourcesSizes().
+	st, err := NewStatusLine(ctx, resources)
+	assert.Nil(t, err)
+
+	// Test GetStatusString() and Increment().
+	st.Start(false)
+	assert.Equal(t, "\r-[                    ]          0/1000 Resources          0 B / 6.0 kB          0s elapsed", st.GetStatusString())
+	st.Increment(0)
+	time.Sleep(10 * time.Millisecond) // Give StatusLine loop time to update fields.
```

Comment:
This is an arbitrary length of time that worked for me on my machine. Alternatively, TestGetStatusString could create and pass a channel that the SL could use to tell TestGetStatusString it has updated its internal states/fields and is ready to be checked -- Sleep would be replaced with a blocking select-case statement. I would assume this would be much more reliable than these Sleep calls